bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gpi/GpiCB.java

```diff
@@ -44,50 +44,46 @@ protected void thisReset(Map<String, Object> options) {
         _sysConfig = dataObj.getSysConfig();
     }
 
+    @Override
     protected boolean thisHasConfiguration() {
         if (_sysConfig == null)
             return false;
         return (_sysConfig.getParameterCount() > 0);
     }
 
-    protected void thisApplyNext(final IConfig config, IConfig prevFull) {
+    @Override
+    protected void thisApplyNext(final IConfig config, final IConfig prevFull) {
         final String systemName = _sysConfig.getSystemName();
-        Collection<IParameter> sysConfig = _sysConfig.getParameters();
-
-        final class MagnitudeFilter implements Function1<Magnitude, Boolean> {
-            private final MagnitudeBand band;
-
-            private MagnitudeFilter(MagnitudeBand band) {
-                this.band = band;
-            }
-
-            @Override
-            public Boolean apply(Magnitude magnitude) {
-                return magnitude.band().equals(band);
-            }
-        }
+        final Collection<IParameter> sysConfig = _sysConfig.getParameters();
 
-        final class MagnitudeSetter implements ApplyOp<Magnitude> {
-            private final String magProp;
-
-            private MagnitudeSetter(String magHProp) {
-                this.magProp = magHProp;
-            }
-
-            @Override
-            public void apply(Magnitude magnitude) {
-                config.putParameter(systemName,
-                            DefaultParameter.getInstance(magProp,
-                                magnitude.value()));
-            }
-        }
-        for (IParameter param : sysConfig) {
+        for (final IParameter param : sysConfig) {
             config.putParameter(systemName,
                     DefaultParameter.getInstance(param.getName(), param.getValue()));
         }
         config.putParameter(systemName,
-                            StringParameter.getInstance(InstConstants.INSTRUMENT_NAME_PROP,
-                                    Gpi.INSTRUMENT_NAME_PROP));
+                StringParameter.getInstance(InstConstants.INSTRUMENT_NAME_PROP,
+                        Gpi.INSTRUMENT_NAME_PROP));
+
+        // Set the magnitude properties.
+        final ISPObsComponent targetcomp = SPTreeUtil.findTargetEnvNode(getObsComponent().getContextObservation());
+        if (targetcomp != null) {
+            final TargetObsComp toc = (TargetObsComp) targetcomp.getDataObject();
+            if (toc != null) {
+                // The Asterism here should be Single(base), so we only need to concern ourselves with the head target.
+                toc.getAsterism().allSpTargetsJava().headOption().foreach(base -> {
+                    final ImList<Magnitude> magnitudes = base.getMagnitudesJava();
+                    MAG_PROPERTIES.forEach((band, propName) ->
+                        magnitudes.find(m -> m.band().equals(band)).
+                                foreach(mag -> config.putParameter(systemName, DefaultParameter.getInstance(propName, mag))));
```

Comment:
Haven't we changed the type of the object that we're storing in the sequence?  Before it was

```java
    config.putParameter(systemName, DefaultParameter.getInstance(magProp, magnitude.value()));
```

which is a `Double` and now it is the `Magnitude` instance itself?

Reply:
Having the `Magnitude` is objectively better than a `Double`, and I don't think the property is used anywhere else except for the seqexec.  But how does the sequence XML compare from the 2017B OT and the OT with this update?  We need to make sure that what we send to the seqexec doesn't change.

Reply:
I'll double-check this.

Reply:
Yes, you are right. We were writing a `Double` and now a `Magnitude`.

I think for now, we should just write the `Double` so we can push out the test release that Bryan wants and if we want to change this later, we can do so. What do you think?